### PR TITLE
Last dialogs

### PR DESCRIPTION
**Pull request by orangejenny**
get rid of the last of the jQuery UI dialogs

@proteusvacuum

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -1091,42 +1051,15 @@ define([
                     msg = "Parsing Error. Please check that your form is valid XML.";
                 }
 
-                // this button doesn't seem to actually exist
-                // todo: fix
```
Comment (orangejenny) [inline src/core.js]: this is a Mike White leftover, happy to put it back if someone else knows what it's talking about...

File: src/core.js [hunk 2]
```diff
@@ -1848,10 +1789,10 @@ define([
         var url = saveType === 'patch' ?  opts.patchUrl : opts.saveUrl;
 
         $(document).ajaxStart(function () {
-            _this.showWaitingDialog();
+            _this.showWaitingModal();
         });
         $(document).ajaxStop(function () {
-            _this._hideConfirmDialog();
+            _this.$f.find('.fd-modal-generic-container .modal').modal("hide");
```
Comment (millerdev) [inline src/core.js]: Why not `_this.closeModal()`?

File: src/javaRosa.js [hunk 3]
```diff
@@ -2102,12 +2102,12 @@ define([
...
                     name: "Edit Bulk Translations",
                     action: function (done) {
-                        _this.showItextDialog(done);
+                        _this.showItextModal(done);
                     }
                 }
             ]);
         },
-        showItextDialog: function (done) {
+        showItextModal: function (done) {
```
Comment (millerdev) [inline src/javaRosa.js]: Curious why you decided to rename from "Dialog" to "Modal". *Modal* is short for *modal dialog*, right?

File: src/core.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ define([
     'jquery.jstree',
     'jquery.bootstrap',
     'jquery.fancybox',  // only thing we use fancybox for is its spinner, no actual display of anything
-    'jquery-ui',  // used for buttons in Edit Source XML, and dialogs
+    'jquery-ui',  // used for autocomplete
```
Comment (millerdev) [inline src/core.js]: is it going to be possible to substitute the autocomplete functionality as well, or are we stuck with jquery-ui?

Comment (orangejenny) [inline src/core.js]: Hopefully we can replace autocomplete with bootstrap's "typeahead" - haven't looked into it yet, but planning on it.

File: src/javaRosa.js (hunk 3 above)
Comment (orangejenny) [inline src/javaRosa.js]: No good reason. For the most part, we've ended up using "dialog" for functions that use the jQuery UI widget and "modal" for the bootstrap widget, so my todo list for this PR was basically "grep -i dialog", and I ended up changing this.

File: src/core.js (hunk 2 above)
Comment (orangejenny) [inline src/core.js]: good point, will fix

File: src/core.js (hunk 4 above)
Comment (emord) [inline src/core.js]: This works with inputs, but unfortunately won't work with content editable divs.

Took a quick look and it seems like there are plugins that handle that, but I'm not sure how well those will work.

**Comment by emord** [comment]
Looks good to me, but can't merge.

**Pull request merged**